test/client_test.js

```diff
@@ -1,430 +1,365 @@
+    t.equal('This is an expected error.', err.message);
     callbn1 += 1;
+    done();
   });
 
-  client.set('foo', 'bar', function(err, val) {
-    if (err) {
-      errn2 += 1;
-    }
+  client.set('foo', 'bar', function(err /*, val */) {
+    t.ok(err, 'Ensure callback called with error');
+    t.equal('This is an expected error.', err.message);
     callbn2 += 1;
-  });
-
-  beforeExit(function() {
-    assert.equal(4, n,  'Ensure set is retried once ' + n);
-    assert.equal(1, callbn1,  'Ensure callback is called ' + callbn1);
-    assert.equal(1, errn1,  'Ensure callback called with error ' + errn1);
-    assert.equal(1, callbn2,  'Ensure callback is called ' + callbn2);
-    assert.equal(1, errn2,  'Ensure callback called with error ' + errn2);
-  });
-}
-
-exports.testSetUnicode = function(beforeExit, assert) {
+    done();
+  });
+
+  var done =(function() {
+    var called = 0;
+    return function() {
+      called += 1;
+      if (called < 2) return; 
+      t.equal(2, n,  'Ensure set is retried');
+      t.equal(1, callbn1,  'Ensure callback is called once');
+      t.equal(1, callbn2,  'Ensure callback is called once');
+      process.nextTick(function() {
+        t.equal(4, n,  'Ensure set is retried'); // not sure what's up here..
+        t.end();
+      });
+    };
+  })();
+});
+
+test('SetUnicode', function(t) {
   var n = 0;
-  var callbn = 0;
   var dummyServer = new MemJS.Server();
   dummyServer.write = function(requestBuf) {
-    request = MemJS.Utils.parseMessage(requestBuf);
-    assert.equal('hello', request.key);
-    assert.equal('éééoào', request.val);
+    var request = MemJS.Utils.parseMessage(requestBuf);
+    t.equal('hello', request.key.toString());
+    t.equal('éééoào', request.val.toString());
     n += 1;
     dummyServer.respond({header: {status: 0, opaque: request.header.opaque}});
-  }
+  };
 
   var client = new MemJS.Client([dummyServer]);
   client.set('hello', 'éééoào', function(err, val) {
-    assert.equal(true, val);
-    callbn += 1;
-  });
-
-  beforeExit(function() {
-    assert.equal(1, n,  'Ensure set is called');
-    assert.equal(1, callbn,  'Ensure callback is called');
+    t.equal(true, val);
+    t.equal(1, n,  'Ensure set is called');
+    t.end();
   });
-}
+});
 
-exports.testAddSuccessful = function(beforeExit, assert) {
+test('AddSuccessful', function(t) {
   var n = 0;
-  var callbn = 0;
   var dummyServer = new MemJS.Server();
   dummyServer.write = function(requestBuf) {
-    request = MemJS.Utils.parseMessage(requestBuf);
-    assert.equal('hello', request.key);
-    assert.equal('world', request.val);
-    assert.equal('\0\0\0\0\0\0\4\0', request.extras.toString());
+    var request = MemJS.Utils.parseMessage(requestBuf);
+    t.equal('hello', request.key.toString());
+    t.equal('world', request.val.toString());
+    t.equal('\0\0\0\0\0\0\4\0', request.extras.toString()); // whaaa?
+    t.equal('0000000000000400', request.extras.toString('hex'));
```

Comment:
Why not? The regular string (line 220) and hex string (line 221) are equivilant.